### DISCUSSION TITLE
helm-dependency with configMaps

### DISCUSSION THREAD

**Discussion by om3171991** [argoproj/argo-cd#8544] [Q&A]
Hi Team,

We are trying to deploy an application using https://github.com/argoproj/argocd-example-apps/tree/master/helm-dependency, but we have a use-case where the base helm dir contains another config dir with properties file and these are required by the dependent helm chart to create config maps, but when I deploy the app via Argo, it creates empty config maps.

**service2 configMap manifest:**
apiVersion: v1
kind: ConfigMap
metadata:
  name: config
data:
  {{- (.Files.Glob "configmap/*").AsConfig | nindent 2 }}

**helmChart dir:**

helm/
  configmap/app.properties
  chart.yaml
  values.yaml

**chart.yaml**

apiVersion: v2
name: service1
description: A Helm chart for Kubernetes
type: application
version: 0.0.1
dependencies:
  - name: service2
    version: 0.1.1
    repository: http://chartmuseum.com/

Need help on how can we make the above thing work.

**Comment by om3171991** [comment]
@alexmt @alexec @jessesuen @mayzhang2000 , Guys can you please help with the above query?

**Reply by steinarox** [reply]
does it work locally when doing `helm template`?

**Reply by om3171991** [reply]
Yes.

**Reply by steinarox** [reply]
argo should print out the full helm template cmd, does that one still work locally?

**Reply by om3171991** [reply]
@steinarox - not sure about yr last comment. Can you please guide me through this?

**Reply by steinarox** [reply]
@om3171991 in argocd UI in the application it should have an ERROR at the top of the screen. If you click that one you should get more details about the errormessage. It should contain the helm template with all the arguments argocd uses for your application. 
![image](https://user-images.githubusercontent.com/93972947/154899407-460cf234-0a97-4362-81e6-5a53af2ec2f3.png)

**Reply by om3171991** [reply]
@steinarox Thanks, but this wasn't related to Argo. There is an open issue with the helm for the same - https://github.com/helm/helm/issues/1892. For now, by shuffling manifest files under the helm chart - able to solve it.
